File: android/app/src/main/java/com/feriadapp/app/WidgetBridgePlugin.java
package com.feriadapp.app;

import android.content.SharedPreferences;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "WidgetBridge")
public class WidgetBridgePlugin extends Plugin {

    private static final String PREFS_NAME = "CapacitorStorage";

    @PluginMethod
    public void updateData(PluginCall call) {
        String days = call.getString("days");
        String label = call.getString("label");
        String title = call.getString("title");
        String name = call.getString("name");
        String isToday = call.getString("isToday", "0");

        if (days == null || label == null || title == null || name == null) {
            call.reject("Missing required fields");
            return;
        }

        SharedPreferences prefs = getContext().getSharedPreferences(PREFS_NAME, android.content.Context.MODE_PRIVATE);
        prefs
            .edit()
            .putString("widget_days", days)
            .putString("widget_label", label)
            .putString("widget_title", title)
            .putString("widget_name", name)
            .putString("widget_is_today", isToday)
            .apply();

        FeriadappWidgetProvider.updateAll(getContext());
        call.resolve();
    }

    @PluginMethod
    public void refresh(PluginCall call) {
        FeriadappWidgetProvider.updateAll(getContext());
        call.resolve();
    }
}

File: ios/App/App/WidgetBridgePlugin.swift
import Foundation
import Capacitor
import WidgetKit

@objc(WidgetBridgePlugin)
public class WidgetBridgePlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "WidgetBridgePlugin"
    public let jsName = "WidgetBridge"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "updateData", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "refresh", returnType: CAPPluginReturnPromise),
    ]

    @objc func updateData(_ call: CAPPluginCall) {
        guard
            let days = call.getString("days"),
            let label = call.getString("label"),
            let title = call.getString("title"),
            let name = call.getString("name")
        else {
            call.reject("Missing required fields")
            return
        }

        let isToday = call.getString("isToday") ?? "0"
        WidgetDataStore.save(days: days, label: label, title: title, name: name, isToday: isToday)

        if #available(iOS 14.0, *) {
            WidgetCenter.shared.reloadAllTimelines()
        }

        call.resolve()
    }

    @objc func refresh(_ call: CAPPluginCall) {
        if #available(iOS 14.0, *) {
            WidgetCenter.shared.reloadAllTimelines()
        }
        call.resolve()
    }
}
